### DISCUSSION TITLE
What all values are supported in shared map?

### DISCUSSION THREAD

**Discussion by anibansal21** [microsoft/FluidFramework#10288] [Q&A]
Can we store shared string as a value in shared map or shared directory? Currently all my DDS are attached to root directory and I want to categorize them, So I was thinking to create different shared directory and in one shared directory store related DDS. All the shared directories would be attached to root directory.

Would there be any difference in eventing also? I mean to say if a shared string is attached to root directory or it is attached to shared directory then eventing mechanism would be same or different?

**Comment by tylerbutler** [comment]
Great questions! You can store references DDSes in other DDSes via their handle. See the documentation here for more info and examples: <https://fluidframework.com/docs/build/data-modeling/#using-handles-to-store-and-retrieve-shared-objects>

Regarding the eventing, it's the same. Using your example, the SharedString events will work the same way regardless of where the DDS reference is stored.

Note, though, that if the SharedString changes, a valueChanged event will _not_ be raised on the SharedMap. This is consistent with how references to plain Objects work -- changes to the underlying object won't cause a valueChanged event. While it's consistent behavior, it may not be what you expect.
